test/tds.js

```diff
@@ -1551,4 +1551,273 @@ describe('generateTdsRuleset', () => {
             ]
         })
     })
+
+    it('should include rules with known custom actions', async () => {
+        const blockList = emptyBlockList()
+        addDomain(blockList, 'default-ignore.invalid', 'Default Ignore entity', 'ignore', [
+            {
+                rule: 'default-ignore\\.invalid\\/known-action-1'
+            },
+            {
+                rule: 'default-ignore\\.invalid\\/block-ctl-fb-1',
+                action: 'block-ctl-fb'
+            },
+            {
+                rule: 'default-ignore\\.invalid\\/known-action-3',
+                exceptions: { types: ['script'] }
+            },
+            {
+                rule: 'default-ignore\\.invalid\\/block-ctl-yt-2',
+                action: 'block-ctl-yt',
+                exceptions: { types: ['script'] }
+            }
+        ])
+        addDomain(blockList, 'default-block.invalid', 'Default Block entity', 'block', [
+            {
+                rule: 'default-block\\.invalid\\/known-action-1'
+            },
+            {
+                rule: 'default-block\\.invalid\\/block-ctl-fb-1',
+                action: 'block-ctl-fb'
+            },
+            {
+                rule: 'default-block\\.invalid\\/known-action-2',
+                exceptions: { types: ['script'] }
+            },
+            {
+                rule: 'default-block\\.invalid\\/block-ctl-yt-2',
+                action: 'block-ctl-yt',
+                exceptions: { types: ['script'] }
+            }
+        ])
+        addDomain(blockList, 'default-custom.invalid', 'Default custom action entity', 'block-ctl-fb', [
+            {
+                rule: 'default-custom\\.invalid\\/custom-default-action-1'
+            },
+            {
+                rule: 'default-custom\\.invalid\\/block-ctl-fb-1',
+                action: 'block-ctl-fb'
+            },
+            {
+                rule: 'default-custom\\.invalid\\/custom-default-action-2',
+                exceptions: { types: ['script'] }
+            },
+            {
+                rule: 'default-custom\\.invalid\\/block-ctl-yt-2',
+                action: 'block-ctl-yt',
+                exceptions: { types: ['script'] }
+            }
+        ])
+
+        await rulesetEqual(blockList, isRegexSupportedTrue, null, {
+            expectedRuleset: [{
```

Comment:
Note that currently these tests capture the actual behavior, will spend a bit more time tomorrow thinking through whether the expected behavior

Reply:
Ok I think the tests are worth a look now too